src/lib/fragments/AgGrid.react.js

```diff
@@ -349,6 +386,10 @@ export default class DashAgGrid extends Component {
             gridColumnApi: params.columnApi,
         });
 
+        if (this.props.rowData) {
+            this.state.rowData = JSON.parse(JSON.stringify(this.props.rowData))
```

Comment:
I see a lot of these cycle-through-JSON operations, I'd like to see if we can avoid them. The ones involving row data will be very expensive for large tables. I guess the issue is ag grid mutating the `rowData` object on user edits... check out https://github.com/ag-grid/ag-grid/issues/3230 and the StackOverflow Q&A in there for a different approach that I think will avoid the need for this deepcopy operation.

Reply:
I'll see what we can do. The issue I think with this is what happens when a custom valueSetter gets added. Also, I think we still have the issue of multiple manipulations.

If we can figure out a better way to copy the old data, we wont need this. And maybe that's what we should do, using componentDidUpdate, when I tried initially it was updating too fast I think.